careerfairsystems/arkad-app-2018
# Optimize API

## Change description

Optimise the API by doing the following:
- If companies are loaded once they won't be loaded again with a loading view
- If a company has been updated all companies will reload

## How to verify

- Verify that when starting the app the first time a loading view* is shown
  - Verify that a error message is shown and a reload button if the API URL is incorrect
- Verify that no loading view* is shown if the companies has been loaded once
  - Verify that no error message is shown when opening the app with a incorrect API URL
  - Verify that updating the companies (by drag and release on the companies screen) generates a error message when having a incorrect API URL
- Verify that companies are loaded when one is updated in Jexpo
  - To do this go to http://dashboard.jexpo.se/arkad/#/login and login (Daniel can give you access). Then navigate to the company ARKADTest2 and change something. Now background and foreground the app
  - You can see that the companies are loading by checking if the status bar loading indicator is visible on iOS. You can also see it by checking the companies screen on both iOS and Android. The drag and release loading indicator will be shown there
  - Also try this with a incorrect API url and verify that a error message is shown

*with loading view I mean the view that is faded dark, not the white one (that is just redux loading)

This is quite a big and important PR so in addition to the above spend some time with some basic regression testing

## Issues fixed

This fixes #60.
